static_frame/core/frame.py

```diff
@@ -6877,63 +6886,67 @@ def unset_columns(self, *,
             names: An sequence of hashables to be used to name the unset columns. If an ``Index``, a single hashable should be provided; if an ``IndexHierarchy``, as many hashables as the depth must be provided.
             index_constructors:
         '''
-        if not names:
-            names = self._columns.names
+        if drop is True and names:
+            raise RuntimeError("The `names` parameter cannot be used with `drop=True` because the column labels will not be included in the resulting Frame.")
 
-        # columns blocks are oriented as "rows" here, and might have different types per row; when moved on to the frame, types will have to be consolidated "vertically", meaning there is little chance of consolidation. A maximal decomposition might give a chance, but each ultimate column would have to be re-evaluated, and that would be expense.
+        if not names and drop is False:
+            names = self._columns.names
 
-        blocks = TypeBlocks.from_blocks(
+        if drop is True:
+            index, own_index, blocks = self._index, self.STATIC, self._blocks
```

Comment:
As Index on the rows are always immutable, here you can assume owning the index.
```suggestion
            index, own_index, blocks = self._index, True, self._blocks
```